apps/system/js/core.js

```diff
@@ -50,6 +54,80 @@
     },
 
     _start: function() {
+      ScreenManager && ScreenManager.turnScreenOn();
+      // We need to be sure to get the focus in order to wake up the screen
+      // if the phone goes to sleep before any user interaction.
+      // Apparently it works because no other window
+      // has the focus at this point.
+      window.focus();
+      // With all important event handlers in place, we can now notify
+      // Gecko that we're ready for certain system services to send us
+      // messages (e.g. the radio).
+      // Note that shell.js starts listen for the mozContentEvent event at
+      // mozbrowserloadstart, which sometimes does not happen till
+      // window.onload.
+      this.publish('mozContentEvent', {
+        type: 'system-message-listener-ready'
+      }, true);
+
+      this.loadWhenIdle([
+        'CameraTrigger',
+        'NotificationScreen',
+        'AirplaneMode',
+        'NotificationsSystemMessage',
+        'Accessibility',
+        'AlarmMonitor',
+        'DebuggingMonitor',
+        'TimeCore',
+        'GeolocationCore',
+        'TetheringMonitor',
+        'UsbCore',
+        'TextSelectionDialog',
+        'ExternalStorageMonitor',
+        'DeviceStorageWatcher',
+        'AppUsageMetrics',
+        'CellBroadcastSystem', // Blocked by integration test.
+        'CpuManager',
+        'HomeGesture',
+        'SourceView',
+        'TtlView',
+        'MediaRecording',
+        'QuickSettings',
+        'UsbStorage',
+        'MobileIdManager',
+        'FindmydeviceLauncher',
+        'FxAccountsManager',
+        'FxAccountsUI',
+        'NetworkActivity',
+        'CrashReporter',
+        'Screenshot',
+        'SoundManager',
+        'CustomDialogService',
+        'CarrierInfoNotifier'
+        // XXX: We should move this into mobileConnectionCore,
+        // but integration tests running on desktop without mobileConnection
+        // is testing this.
+      ]).then(function() {
+        this.startAPIHandlers();
+        return Promise.resolve();
+      }.bind(this)).then(function() {
+        return LazyLoader.load([
+          'js/download/download_manager.js',
+          'js/payment.js',
+          'js/identity.js',
+          'js/devtools/logshake.js',
+          'js/devtools/remote_debugger.js',
+          'js/devtools/developer_hud.js',
+          'shared/js/date_time_helper.js'
+        ]);
+      }.bind(this)).then(function() {
+        this.remoteDebugger = new RemoteDebugger();
+        this.developerHud = new DeveloperHud();
+        this.developerHud.start();
+      }.bind(this));
+    },
+
+    startAPIHandlers: function() {
       for (var api in this.REGISTRY) {
         this.debug('Detecting API: ' + api +
           ' and corresponding module: ' + this.REGISTRY[api]);
```

Comment:
Moreover, follow what I've asked below, if BaseModule could accept a literal object as the extended method lists, it looks that we could keep current convention as:

   var SomeClass = function() {};
   SomeClass.prototype.method = function() {
     this.foo = something;
   };
   BaseModule.create(SomeClass.prototype)